src/avram/charms/enum.cr

```diff
@@ -0,0 +1,69 @@
+macro avram_enum(enum_name, &block)
+  enum {{ enum_name }}
+    {{ block.body }}
+  end
+
+  class Avram{{ enum_name }}
+    def self.adapter
+      Lucky
+    end
+
+    # You may need to prefix with {{ @type }}
+    #
+    #   {{ @type }}::{{enum_name}}
+    def initialize(@enum : {{ enum_name }})
+    end
+
+    def initialize(enum_value : Int32)
+      @enum = {{ enum_name }}.from_value(enum_value)
+    end
+
+    def initialize(enum_value : String)
+      @enum = {{ enum_name }}.from_value(enum_value.to_i)
+    end
+
+    def to_s
+      @enum.to_s
+    end
+
+    def value
+      @enum
+    end
+
+    def blank?
+      @enum.nil?
+    end
+
+    module Lucky
```

Comment:
The enum *wrapper* can be simplified thus:

```suggestion
  struct Avram{{ enum_name }}
    def self.adapter
      Lucky
    end

    # You may need to prefix with {{ @type }}
    #
    #   {{ @type }}::{{enum_name}}
    def initialize(@enum : {{ enum_name }})
    end

    def initialize(enum_value : Int32)
      @enum = {{ enum_name }}.from_value(enum_value)
    end

    def initialize(enum_value : String)
      @enum = {{ enum_name }}.from_value(enum_value.to_i)
    end

    forward_missing_to @enum

    module Lucky
```
Struct, because they are a better fit for immutable type wrappers.

Forwarding to `@enum` means one can do:
```crystal
status.opened? # <= instead of `status.value.opened?`
status.value # <= instead of `status.value.value` (`.value` called twice)
```

Reply:
Really nice, fixed this. Thanks @akadusei